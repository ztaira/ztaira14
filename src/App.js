import React, { Component } from 'react';
import { BrowserRouter, Route, Redirect } from "react-router-dom";
import './App.css';
import SideBar from "./SideBar/SideBar.js";
import MenuBar from './MenuBar/MenuBar.js';
import Homepage from "./Content/Homepage.js";
import PageNotFound from "./Content/PageNotFound.js";
import ModularItemPage from "./Content/ModularItemPage.js";
import Page from "./Content/Page.js";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      menuitems: [
        'home',
        'thing2',
        'thing3',
        'thing4',
      ],
      sidebaritems: [
        {'label': "lol", 'link': ''},
        {'label': "foo1", 'link': 'https://github.com/ztaira14'},
        {'label': "foo2", 'link': 'https://www.google.com'},
        {'label': "foo3", 'link': 'home'},
      ],
      modularitems: [
        "Hello world",
        "Hello world again",
        "blahblahblahblahblahblah lol",
        "Man this is some ugly site",
        "I really hope this gets better",
        "one more",
        "one moore",
        "one mooore",
        "and another",
        "and another!",
      ],
    }
  }

  render() {
    return (
      <BrowserRouter>
        <div className="App">
          <Redirect from="/" to="/home" />
          <SideBar menuitems={this.state.sidebaritems}/>
          <MenuBar menuitems={this.state.menuitems}/>
          <Route path="/home" component={Homepage} />
          <Route path="/thing2" component={PageNotFound} />
          <Route path="/thing3" component={() => <ModularItemPage items={this.state.modularitems} />} />
          <Route path="/thing4" component={() => <Page url="https://raw.githubusercontent.com/ztaira14/abacus/master/README.md" />} />
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
