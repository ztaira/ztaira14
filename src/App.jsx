import React, { Component } from 'react';
import { Route, browserHistory, HashRouter } from 'react-router-dom';
import './App.css';
import SideBar from './SideBar/SideBar.jsx';
import MenuBar from './MenuBar/MenuBar.jsx';
import Homepage from './Content/Homepage.jsx';
import ProjectsPage from './Content/ProjectsPage.jsx';
import ModularListPage from './Content/ModularListPage.jsx';
import Page from './Content/Page.jsx';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      menuitems: [
        'Home',
        'Projects',
        'Essays',
      ],
      sidebaritems: [
        {
          'label': 'GitHub',
          'link': 'https://github.com/ztaira14',
          'image': 'https://simpleicons.org/icons/github.svg',
        },
        {
          'label': 'LinkedIn',
          'link': 'https://www.linkedin.com/in/zacharytaira/',
          'image': 'https://simpleicons.org/icons/linkedin.svg',
        },
      ],
      allEntries: [],
    }
  }

// ===========================================================================
// GET THE REPOSITORY DATA FROM GITHUB
// ===========================================================================

  componentDidMount() {
    this.GetText('https://raw.githubusercontent.com/ztaira14/journal/master/index.json');
  }

  componentWillUnmount() {
    this.xhp.abort();
  }

  GetText(url) {
    let _this = this;
    this.xhp = new XMLHttpRequest();
    this.xhp.onload = function() {
      _this.setState({
        allEntries: JSON.parse(this.responseText),
      });
    };
    this.xhp.open('GET', url);
    this.xhp.send();
  }

  ReturnListRoutes(entry) {
    let url = 'https://raw.githubusercontent.com/ztaira14/journal/master/';
    return (
      <Route
        key={entry.file_name}
        path={"/Essays/" + entry.file_name.slice(0, -3)}
        component={() => <Page
          url={url + entry.file_name}
        />}
      />)
  }

  render() {
    return (
      <HashRouter history={browserHistory}>
        <div className="App">
          <SideBar menuitems={this.state.sidebaritems} />
          <MenuBar menuitems={this.state.menuitems} />
          <Route exact path="/" component={Homepage} />
          <Route path="/Home" component={Homepage} />
          <Route path="/Projects" component={ProjectsPage} />
          <Route
            exact
            path="/Essays"
            component={() => <ModularListPage
              allEntries={this.state.allEntries}
            />}
          />
          {this.state.allEntries.map(this.ReturnListRoutes)}
        </div>
      </HashRouter>
    );
  }
}

export default App;
