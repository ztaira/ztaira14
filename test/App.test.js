import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../src/App.jsx';

beforeEach(function() {
  // to mock localStorage
  var localStorageMock = (function() {
    var store = {};
    return {
      getItem: function(key) {
        return store[key];
      },
      setItem: function(key, value) {
        store[key] = value.toString();
      },
      clear: function() {
        store = {};
      },
      removeItem: function(key) {
        delete store[key];
      }
    };
  })();
  Object.defineProperty(window, 'localStorage', { value: localStorageMock });
});

it('has not changed since the last snapshot', () => {
  const app = renderer.create(
    <App />
  );
  let app_json = app.toJSON();
  expect(app_json).toMatchSnapshot();
});

it('has the expected menu items', () => {
  const app = ReactDOM.render(
    <App />,
    document.body.appendChild(document.createElement('div'))
  );
  expect(app.state.menuitems).toEqual(['Home', 'Projects']);
});

it('has the expected sidebar items', () => {
  const app = ReactDOM.render(
    <App />,
    document.body.appendChild(document.createElement('div'))
  );
  expect(app.state.sidebaritems).toEqual(
    [
        {
          'label': 'GitHub',
          'link': 'https://github.com/ztaira14',
          'image': 'https://simpleicons.org/icons/github.svg',
        },
        {
          'label': 'LinkedIn',
          'link': 'https://www.linkedin.com/in/zacharytaira/',
          'image': 'https://simpleicons.org/icons/linkedin.svg',
        },
    ]);
});
