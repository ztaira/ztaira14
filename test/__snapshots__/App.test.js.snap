exports[`test has not changed since the last snapshot 1`] = `
<div
  className="App">
  <ul
    className="MenuBar">
    <li>
      <a
        href="#/Home"
        onClick={[Function]}>
        Home
      </a>
    </li>
    <li>
      <a
        href="#/Projects"
        onClick={[Function]}>
        Projects
      </a>
    </li>
    <li>
      <a
        href="#/Blog"
        onClick={[Function]}>
        Blog
      </a>
    </li>
  </ul>
  <div
    className="HomePage">
    <div
      className="HomePageContent">
      <div
        className="Bio">
        <img
          alt=""
          className="ProfilePic"
          src="https://avatars0.githubusercontent.com/u/6549364?v=3" />
        <h1>
          Zachary Taira
        </h1>
        <p>
          zach.taira@gmail.com
        </p>
        <p>
          BS/MS at NEU
        </p>
        <div
          className="RightColumnSpacer" />
      </div>
      <div
        className="Intro">
        <p>
          Hello, and welcome!
        </p>
        <p>
          My name is Zachary Taira. I\'m an EE student at Northeastern University, and I built this site in order to document my coding journey. You can see some of my other projects under the Projects page, and can read some of my thoughts and writeups under the Blog page.
        </p>
        <p>
          If you want some more info about this site, it\'s Hack-A-Week 19 under the Projects page. Happy browsing!
        </p>
        <p>
          Zach
        </p>
      </div>
    </div>
  </div>
</div>
`;
